avocado/plugins/diff.py

```diff
@@ -0,0 +1,422 @@
+        self.std_diff_output = True
+
+    def configure(self, parser):
+        """
+        Add the subparser for the diff action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Diff, self).configure(parser)
+
+        parser.add_argument("jobids",
+                            default=[], nargs=2,
+                            metavar="<JOB>",
+                            help='A job reference, identified by a (partial) '
+                            'unique ID (SHA1) or test results directory.')
+
+        parser.add_argument('--html', type=str,
+                            metavar='FILE',
+                            help='Enable HTML output to the FILE where the '
+                            'result should be written.')
+
+        parser.add_argument('--open-browser',
+                            action='store_true',
+                            default=False,
+                            help='Generate and open a HTML report in your '
+                            'preferred browser. If no --html file is '
+                            'provided, create a temporary file.')
+
+        parser.add_argument('--diff-filter',
+                            dest='diff_filter',
+                            type=self._validate_filters,
+                            default=['cmdline', 'time', 'variants',
+                                     'results', 'config', 'sysinfo'],
+                            help='Comma separated filter of diff sections: '
+                            '(no)cmdline,(no)time,(no)variants,(no)results,\n'
+                            '(no)config,(no)sysinfo (defaults to all '
+                            'enabled).')
+
+        parser.add_argument('--paginator',
+                            choices=('on', 'off'), default='on',
+                            help='Turn the paginator on/off. '
+                            'Current: %(default)s')
+
+        parser.add_argument('--create-reports', action='store_true',
+                            help='Create temporary files with job reports '
+                            '(to be used by other diff tools)')
+
+        parser.epilog = 'By default, a textual diff report is generated '\
+                        'in the standard output.'
+
+    def run(self, args):
+        job1_dir, job1_id = self._setup_job(args.jobids[0])
+        job2_dir, job2_id = self._setup_job(args.jobids[1])
+
+        job1_data = self._get_job_data(job1_dir)
+        job2_data = self._get_job_data(job2_dir)
+
+        report_header = 'Avocado Job Report\n'
+        job1_results = [report_header]
+        job2_results = [report_header]
+
+        if 'cmdline' in args.diff_filter:
+            cmdline1 = self._get_command_line(job1_dir)
+            cmdline2 = self._get_command_line(job2_dir)
+
+            if str(cmdline1) != str(cmdline2):
+                command_line_header = ['\n',
+                                       '# COMMAND LINE\n']
```

Comment:
Followup to https://github.com/avocado-framework/avocado/pull/1371#discussion_r74442322

I'm unable to distinguish between the sections. It's way too hidden. I'd go with

```
###############################################################################
# COMMAND LINE
################################################################################

COMMAND LINE
=============

=============
COMMAND LINE
=============
```
separated by 2 newlines from top and 1 newline from bottom.

I can accept 1 and 1 newline and I won't block the 1 and 0 newlines (in RST format it ought to be 1 and 1 as === followed by text looks strange).

I really don't like this one-line format as it's unrecognizable to me. I'm wondering what difficulties the multi-line format does to standard tools. I tried `diff` and `meld` and the output looked good.

Last but not least, try uncolored output with multiple tests. Also very hard to search for the sections...